flink-test-utils-parent/flink-test-utils/src/main/java/org/apache/flink/streaming/util/TestStreamEnvironment.java

```diff
@@ -74,30 +74,17 @@ public static void setAsContext(
                     TestStreamEnvironment env =
                             new TestStreamEnvironment(
                                     miniCluster, parallelism, jarFiles, classpaths);
-                    randomize(conf);
+                    if (RANDOMIZE_CHECKPOINTING_CONFIG) {
+                        randomize(
+                                conf, ExecutionCheckpointingOptions.ENABLE_UNALIGNED, true, false);
```

Comment:
Could you please elaborate why the randomization is done non-uniformly?
I guess for backend you need to update the configuration earlier. Does it make sense to do the same for UC?